packages/synapse-react-client/src/utils/hooks/useImmutableTableQuery/useImmutableTableQuery.ts

```diff
@@ -102,6 +102,70 @@ export type UseImmutableTableQueryOptions = {
 // When changing the query with the debounce option, the amount of time to wait for additional debounced changes before updating the query
 export const DEBOUNCE_DELAY_MS = 750
 
+/**
+ * Utility to synchronize the URL with the current query request. Synchronization only occurs if `shouldDeepLink` is true.
+ * @param shouldDeepLink whether the query should be synchronized with the URL
+ * @param componentIndex the index of the component to synchronize with the URL - essential if multiple components to synchronize are on the page
+ * @param setQuery - callback used to update the current query
+ * @param initQueryRequest - the initial query request passed via props
+ * @param currentQueryRequest - the query request that is currently used
+ */
+function useSynchronizeQueryWithUrl(
+  shouldDeepLink: boolean,
+  componentIndex: number,
+  setQuery: (
+    queryRequest: React.SetStateAction<QueryBundleRequest>,
+    commitOptions?: QueryChangeCommitOptions,
+  ) => void,
+  initQueryRequest: QueryBundleRequest,
+  currentQueryRequest: QueryBundleRequest,
+) {
+  /**
+   * Inspect the URL on mount to see if we have a particular query request that we must show.
+   */
+  useEffect(() => {
+    // Only run this effect if deep linking is enabled
+    if (shouldDeepLink) {
+      const queryRequestFromLink = DeepLinkingUtils.getQueryRequestFromLink(
+        'QueryWrapper',
+        componentIndex,
+      )
+      if (queryRequestFromLink && queryRequestFromLink.query) {
+        setQuery(prevState => ({
+          ...prevState,
+          ...queryRequestFromLink,
+          query: {
+            ...prevState.query,
+            ...queryRequestFromLink.query,
+          },
+        }))
+      }
```

Comment:
change: check that `shouldDeepLink` is true before updating the query based on the URL.

The bug I was running into was something like

1. I searched for a specific dataset on the datasets explore page
2. The search would update the URL
3. I would attempt to add the dataset to the download cart
4. [BUG] The inner `QueryWrapper` surrounding the `TableQueryDownloadConfirmation` saw the URL search parameters and updated the query to match the search
5. The TableQueryDownloadConfirmation component would use the explore page query in the instead of the query constructed using the dataset/table synID

Now, step 4 will not happen because `shouldDeepLink` is false.